File: Level-2/17.java
class Solution {
    public int[] dp;
    public int solution(int n) {
        dp = new int[n + 1];
        int answer = fibo(n);
        return answer;
    }
    
    public int fibo(int x){
        if(x == 0 || x == 1) return x;
        if(dp[x] != 0) return dp[x];
        dp[x] = fibo(x - 1) + fibo(x - 2);
        dp[x] %= 1234567;
        return dp[x];
    }
}

File: Level-2/17.js
function solution(arr1, arr2) {
    const answer = Array.from(Array(arr1.length), () => new Array(arr2[0].length).fill(0));
    const n = arr2.length;
    
    for(let i = 0; i < answer.length; i++) {
        for(let j = 0; j < answer[i].length; j++) {
            for(let k = 0; k < n; k++) {
                answer[i][j] += arr1[i][k] * arr2[k][j];
            }
        }
    }
    
    return answer;    
}
